core/services/workflows/engine.go

```diff
@@ -385,58 +440,30 @@ func (e *Engine) registerTrigger(ctx context.Context, t *triggerCapability, trig
 	return nil
 }
 
-// loop is the synchronization goroutine for the engine, and is responsible for:
-//   - dispatching new workers up to the limit specified (default = 100)
-//   - starting a new execution when a trigger emits a message on `triggerEvents`
-//   - updating the `executionState` with the outcome of a `step`.
+// stepUpdateLoop is a singleton goroutine for the engine, and it updates the `executionState` with the outcome of a `step`.
```

Comment:
Can we update the description? It's not a singleton goroutine for the engine, it's a singleton per execution